08-rails/debugging-rails-vscode.md

```diff
@@ -0,0 +1,118 @@
+# Debugging Rails with VS Code
+## Learning Goals
+- Practice setting breakpoints and watch values.
+- Additional experience tracking the flow of logic in a Rails application.
+
+
+## Adding New tools
+VS Code and Ruby are still immature in their compatibility, but with a few extensions we can leverage the debugger that comes with VS Code to make our lives a lot easier.
+
+To start off, we should make sure that we have the [ruby](https://marketplace.visualstudio.com/items?itemName=rebornix.Ruby) VS Code extension installed. Use `⇧⌘X` to open the extension window and check to see that it is currently installed.
+
+Additionally, we need to modify the `Gemfile`:
+
+```ruby
+#Gemfile
+group :development, :test do
+  gem 'ruby-debug-ide'
+  gem 'debase'
+end
+```
+
+The [debase](https://rubygems.org/gems/debase) gem is an implementation of the standard Ruby debugger that works with Ruby 2.0, and [ruby-debug-ide](https://rubygems.org/gems/ruby-debug-ide) gem he helps `debase` communicate with VS Code so that we can see and set breakpoints in the IDE itself.
+
+__NOTE: These gems should ALWAYS be part of development and test ONLY.__
+
+Once they are included in your `Gemfile`, run the following commands:
+
+```ruby
+  bundle install 
+  #installs the new gems we just added to the Gemfile
+
+  $: which bundle
+  # returns /path/to/rubygem/bin/bundle
+
+  $: bundle show ruby-debug-ide
+  #returns /path/to/rubygem/gems/ruby-debug-ide-x.x.x
+
+  $: which rspec
+  #returns /path/to/rubygem/bin/rspec
+```
+
+We will need the output from the commands from above in just a moment in order to tell VS Code how to talk to these gems.
+
+Open the debugging panel using ⇧⌘D, and click on the drop down menu near the play icon. Select `Add New Configuration`.
+
+![alt-text](images/rails-debugger-config-setup.png "The cursor is highlighting the Add New Configuration option on a drop down menu in VS Code's debugging menu")
+
+The next selections you make are somewhat arbitrary, as we will be replacing the code that this action generates with our own custom version.
+
+Once you have the `launch.json` file loaded and open, copy the code below and paste those three paths into the appropriate locations. You should paste each path in two different spots in the file (please remove the asterisks).
+
```

Comment:
In addition to the asterisks, it might be helpful to have a little shortlist of which fields they should be replacing listed before the json they need to copy/paste, so they see this list of what values to go through and check from the beginning:

1. pathToBundler
1. pathToRDebugIDE
1. program
1. pathToBundler
1. pathToRDebugIDE